### PR TITLE
Create/location search UI

### PR DESCRIPTION
**Pull request by ThomasPFMuller**
Finished the logic to add waypoints manually either by searching or by clicking on the map. A minor decisions was taken:
- The path linking all markers of the itinerary will not be reconstructed at every modification of the list of locations. To see the final path, the user must go to the preview screen.

In terms of code coverage, I have worked on some tests for the new composables, but I cannot get them to work for the moment, so code coverage is low. I will add tests for MS3

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreateItineraryMap.kt
Comment (ethangraham2001) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreateItineraryMap.kt:1]: Great stuff. Thanks for cleaning up some of that less readable stuff.

File: app/src/main/java/com/github/wanderwise_inc/app/ui/map/LocationSearchBar.kt
Comment (ethangraham2001) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/map/LocationSearchBar.kt:1]: lgtm

File: app/src/test/java/com/github/wanderwise_inc/app/ui/navigation/CreationNavigationTest.kt [hunk 1]
```diff
@@ -75,6 +75,8 @@ class CreationNavigationTest {
       every { createItineraryViewModel.getPolylinePointsLiveData() } returns dummyLiveData
       every { createItineraryViewModel.getNewItinerary() } returns
           Itinerary.Builder(userUid = "uniqueUserUID")
+      every { createItineraryViewModel.createItineraryManually } returns false
+      every { createItineraryViewModel.createItineraryByTracking } returns false
 
       every { firebaseAuth.currentUser?.uid } returns null
```
Comment (ethangraham2001) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/navigation/CreationNavigationTest.kt:81]: Just in passing as I must have missed it before - In theory we no longer need to mock firebaseAuth as we're going here. All calls are going through profileViewModel now

**Review by ethangraham2001 — APPROVED** [review]
Great work

File: app/src/test/java/com/github/wanderwise_inc/app/model/location/ItineraryTest.kt [hunk 2]
```diff
@@ -79,7 +81,7 @@ class ItineraryTest {
     assertEquals("user123", builder.userUid)
     assertEquals("Changed Title", builder.title)
     assertEquals(2, builder.tags.size) // New one should have 2 tags
-    assertEquals(3, builder.locations.size) // New one should have 3 locations
+    assertEquals(2, builder.locations.size) // New one should have 3 locations
```
Comment (276746) [inline app/src/test/java/com/github/wanderwise_inc/app/model/location/ItineraryTest.kt]: comments deprecated

**Review by 276746 — APPROVED** [review]
All good nice job ! The commented code will need to be corrected when correcting the tests

**Pull request merged**